From: Kurt Roeckx <kurt@roeckx.be>
Date: Mon, 29 Oct 2018 03:16:19 +0100
Subject: Disable the freelist

We don't define OPENSSL_NO_BUF_FREELISTS globally sinc it changes structures and
would break the ABI.  Instead we just do it in the .c files that try to do
something with it.
---
 ssl/s3_both.c | 1 +
 ssl/ssl_lib.c | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/ssl/s3_both.c b/ssl/s3_both.c
index 054ded1..bb0085c 100644
--- a/ssl/s3_both.c
+++ b/ssl/s3_both.c
@@ -584,6 +584,7 @@ int ssl_verify_alarm_type(long type)
     return (al);
 }
 
+#define OPENSSL_NO_BUF_FREELISTS
 #ifndef OPENSSL_NO_BUF_FREELISTS
 /*-
  * On some platforms, malloc() performance is bad enough that you can't just
diff --git a/ssl/ssl_lib.c b/ssl/ssl_lib.c
index aa0cbdb..8875409 100644
--- a/ssl/ssl_lib.c
+++ b/ssl/ssl_lib.c
@@ -162,6 +162,8 @@
 
 const char *SSL_version_str = OPENSSL_VERSION_TEXT;
 
+#define OPENSSL_NO_BUF_FREELISTS
+
 SSL3_ENC_METHOD ssl3_undef_enc_method = {
     /*
      * evil casts, but these functions are only called if there's a library
